### ISSUE TITLE
Feature Request: Dataton Watchout module

### ISSUE THREAD

**Issue by dnmeid** [bitfocus/companion#24]
Add support for Dataton Watchout, a very popular multiscreen presentation system. I'd be glad to help with this.

**Comment by willosof** [comment]
Please help me find some documentation on how the TCP or UDP protocol works. 

From the watchout documentation, I understood there was different control protocols. Is the "Production Computer Control" what you want?

> Production Computer Control (TCP and UDP). Activates external control of the WATCHOUT
> production software. This allows you to control your WATCHOUT presentation from a touch panel or
> other computers and control systems. The control protocol has provisions for positioning, starting and
> stopping the presentation, as well as other miscellaneous functions. See Production Computer
> Protocol.
> 
> NOTE: This control feature is not to be confused with the display cluster control capability, described
> under Control Protocol. While they both perform similar functions, one controls the production computer
> and the other controls the display computers directly, without the need for any production computer
> being present during playback.

Please attach documentation!

**Comment by dnmeid** [comment]
Willosof, at the moment I run a local repo with working watchout branch. I want to give it a try next week on a real show. Unfortunately I have problems keeping my fork up to date before I push the branch to github. Then I'll do a pull request for you.

At the moment I run all the code in one file and don't use events to communicate between the tcp-client and the driver itself. The other drivers like OSC and UDP rely on events, but unlike UDP every instance needs a connection of it's own. That's why I didn't use events. At the moment I also don't use promises, there is so far no mechanism to react on incoming data in relation to outgoing data. Incoming data can be computed of course but only unrelated to sent commands. Maybe you want to put up an prototype for TCP-client based drivers and I make the watchout protocol stuff?

Basically the protocols for production computer and display clusters are different subsets of a bunch of commands. In my shows I'm using the production computer protocol 95% of the time, but implementing display cluster protocol as well isn't a big deal.

**Comment by dnmeid** [comment]
I tested the Watchout module today with a real watchout system, it is working very well, so the issue can be closed. I will refine the module later and also add one for the display cluster, but this module satisfies all of the daily needs.

**Comment by calwalker1** [comment]
Hi all, This module is great at controlling Watchout systems via a production computer, however in some install/theatre shows the controls need to be sent to a display/display cluster.
This would probably need to be a different/new module (as you interact with it slightly differently).

I have a bit of experience with Watchout TCP programming but am not familiar with how modules are built in this. So I guess i'm semi asking for a new display module/offering to make one by repurposing the code on the production PC module.
